pkg/cloud/services/secretsmanager/secret.go

```diff
@@ -33,10 +36,15 @@ const (
 	namespacesPrefix = "namespaces"
 	clustersPrefix   = "clusters"
 	instancesPrefix  = "instances"
+
+	// we set the max secret size to well below the 10240 byte limit, because this is limit after base64 encoding,
+	// but the aws sdk handles encoding for us, so we can't send a full 10240.
+	maxSecretSizeBytes = 7000
 )
 
-// Create stores a secret in AWS Secrets Manager for a given machine
-func (s *Service) Create(m *scope.MachineScope, data []byte) (string, error) {
+// Create stores data in AWS Secrets Manager for a given machine, chunking at 10kb per secret. The prefix of the secret
+// ARN and the number of chunks are returned.
+func (s *Service) Create(m *scope.MachineScope, data []byte) (string, int32, error) {
```

Comment:
This needs more extensive tests, especially around the expected chunks and so on.

Reply:
Resolved by @vincepri breaking out the chunking with its own tests.